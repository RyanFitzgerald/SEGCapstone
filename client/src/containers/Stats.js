import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import Submenu from '../components/stats/Submenu';
import Total from '../components/stats/Total';
import Type from '../components/stats/Type';
import Salesmen from '../components/stats/Salesmen';
import Referral from '../components/stats/Referral';

class Stats extends React.Component {
  constructor() {
    super();

    // Bind functions
    this.setActiveSubtab = this.setActiveSubtab.bind(this);
    
    // State
    this.state = {
      activeSubtab: 1
    }
  }

  componentDidMount() {
    // Set Page Title
    document.title = 'Statistics | Renovaction';

    // Update tab
    this.props.setActiveTab(4);
  }

  setActiveSubtab(tab) {
    this.setState({activeSubtab: tab});
  }

  render() {
    const level = JSON.parse(sessionStorage.getItem('user')).role.level;

    return (
      <div>
        <Submenu activeSubtab={this.state.activeSubtab}/>

        <Switch>
          <Route exact path="/stats" render={() => (
            (level < 2) ? (
              <Redirect to='/'/>
            ) : (
              <Total setActiveSubtab={this.setActiveSubtab}/>
            )
          )}/>
          <Route path="/stats/type" render={() => (
            (level < 2) ? (
              <Redirect to='/'/>
            ) : (
              <Type setActiveSubtab={this.setActiveSubtab}/>
            )
          )}/>
          <Route path="/stats/salesmen" render={() => (
            (level < 2) ? (
              <Redirect to='/'/>
            ) : (
              <Salesmen setActiveSubtab={this.setActiveSubtab}/>
            )
          )}/>
          <Route path="/stats/referral" render={() => (
            (level < 2) ? (
              <Redirect to='/'/>
            ) : (
              <Referral setActiveSubtab={this.setActiveSubtab}/>
            )
          )}/>
        </Switch>
      </div>
    );
  }
}

export default Stats;