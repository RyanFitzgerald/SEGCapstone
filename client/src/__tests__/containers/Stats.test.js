import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Import components
import Stats from '../../Containers/Stats';

// Setup axios mock
const mock = new MockAdapter(axios);

describe('<Stats />', () => {
  beforeAll(() => {
    // Add necessary info to user
    const user = {
      role: {
        level: 1
      }
    };
    sessionStorage.setItem('user', JSON.stringify(user));
  });

  it('should render correctly', () => {
    // Return empty objects on requests
    mock.onAny().reply(200, {});

    // Create mock functions / spies
    const setActiveTabMock = jest.fn();
   
    // Shallow render it
    const wrapper = shallow(
      <Stats setActiveTab={setActiveTabMock}/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Check set active tab is called and with correct value
    expect(setActiveTabMock).toHaveBeenCalledTimes(1);
    expect(setActiveTabMock.mock.calls[0][0]).toBe(4);
  });
});

