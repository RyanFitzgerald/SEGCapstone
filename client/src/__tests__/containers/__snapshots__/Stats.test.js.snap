// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats /> should render correctly 1`] = `
<div>
  <Submenu
    activeSubtab={1}
  />
  <Switch>
    <Route
      exact={true}
      path="/stats"
      render={[Function]}
    />
    <Route
      path="/stats/type"
      render={[Function]}
    />
    <Route
      path="/stats/salesmen"
      render={[Function]}
    />
    <Route
      path="/stats/referral"
      render={[Function]}
    />
  </Switch>
</div>
`;
